leshan-client-core/src/main/java/org/eclipse/leshan/client/engine/DefaultRegistrationEngine.java

```diff
@@ -179,16 +179,28 @@ public void start() {
         stop(false); // Stop without de-register
         synchronized (this) {
             started = true;
-            // Try factory bootstrap
-            // TODO support multi server
-            LwM2mServer dmServer = factoryBootstrap();
-
+            // TODO support multiple servers
+            LwM2mServer dmServer;
+            if (registeredServers.isEmpty()) {
+                // Try factory bootstrap
+                dmServer = factoryBootstrap();
+            } else {
+                dmServer = registeredServers.entrySet().iterator().next().getValue();
+            }
             if (dmServer == null) {
                 // If it failed try client initiated bootstrap
                 if (!scheduleClientInitiatedBootstrap(NOW))
                     throw new IllegalStateException("Unable to start client : No valid server available!");
             } else {
-                registerFuture = schedExecutor.submit(new RegistrationTask(dmServer));
+                // If there exists a registered server already, we try to send a registration update
```

Comment:
I'd actually be sceptical here. If a client is stopped, it should not keep registrations. So this situation should not be allowed. 

If somebody truly needs a LWM2M client that supports this, they can either implement their own registration engine or subclass it.

I am not in favor of doing this.

Reply:
Thx for taking time to look at this and also for sharing your opinion. :pray: 

For now, no plan to integrate this BUT if for some reason we decide to ingrate it  this will not be the default behavior.